code/game/machinery/computer/buildandrepair.dm

```diff
@@ -372,16 +372,31 @@
 			return
 	return
 
-/obj/item/weapon/circuitboard/proc/solder_improve(mob/user as mob)
+/obj/item/weapon/circuitboard/proc/solder_improve(mob/user)
 	to_chat(user, "<span class='warning'>You fiddle with a few random fuses but can't find a routing that doesn't short the board.</span>")
 	return
 
-/obj/item/weapon/circuitboard/supplycomp/solder_improve(mob/user as mob)
+
+/obj/item/weapon/circuitboard/fishtank/solder_improve(mob/user)
+	to_chat(user, "<span class='notice'>You modify the circuitry to support a larger tank.</span>")
+	var/obj/item/weapon/circuitboard/fishwall/A = new /obj/item/weapon/circuitboard/fishwall(src.loc)
+	user.put_in_hands(A)
+	qdel(src)
+	return	
```

Comment:
please remove these trailing returns (a return which comes at the end of a proc where the code already returns by default)